heartbeat/monitors/monitor.go

```diff
@@ -0,0 +1,209 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package monitors
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"sync"
+
+	"github.com/pkg/errors"
+
+	"github.com/elastic/beats/heartbeat/scheduler"
+	"github.com/elastic/beats/heartbeat/watcher"
+	"github.com/elastic/beats/libbeat/beat"
+	"github.com/elastic/beats/libbeat/common"
+	"github.com/elastic/beats/libbeat/logp"
+)
+
+// Monitor represents a configured recurring monitoring task loaded from a config file. Starting it
+// will cause it to run with the given scheduler until Stop() is called.
+type Monitor struct {
+	name           string
+	config         *common.Config
+	registrar      *pluginsReg
+	uniqueName     string
+	scheduler      *scheduler.Scheduler
+	jobTasks       []*monitorTask
+	enabled        bool
+	watchPollTasks []*monitorTask
+	watch          watcher.Watch
+	runMtx         sync.Mutex
```

Comment:
I guess `watchPollTasks`, `watch`, and `runMtx` are all related to dynamic jobs via file config watcher. Let's either distuingish them visually by adding a newline and comment or introduce a type/struct to combine watch related fields.